lib/commands/device/deactivate.ts

```diff
@@ -0,0 +1,111 @@
+/**
+ * @license
+ * Copyright 2020 Balena Ltd.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { flags } from '@oclif/command';
+import type { IArg } from '@oclif/parser/lib/args';
+import { BalenaAmbiguousDevice } from 'balena-errors';
+import Command from '../../command';
+import * as cf from '../../utils/common-flags';
+import { getBalenaSdk, stripIndent } from '../../utils/lazy';
+import type * as BalenaSdk from 'balena-sdk';
+
+interface FlagsDef {
+	yes?: boolean;
+	help: void;
+}
+
+interface ArgsDef {
+	uuid: string;
+}
+
+export default class DeviceDeactivateCmd extends Command {
+	public static description = stripIndent`
+		Deactivate a device.
+
+		Deactivate a device.
+
+		Note this command asks for confirmation interactively.
+		You can avoid this by passing the \`--yes\` option.
+		`;
+	public static examples = [
+		'$ balena device deactivate 7cf02a6',
+		'$ balena device deactivate 7cf02a6 --yes',
+	];
+
+	public static args: Array<IArg<any>> = [
+		{
+			name: 'uuid',
+			description: 'the UUID of the device to be deactivated',
+			required: true,
+		},
+	];
+
+	public static usage = 'device deactivate <uuid>';
+
+	public static flags: flags.Input<FlagsDef> = {
+		yes: cf.yes,
+		help: cf.help,
+	};
+
+	public static authenticated = true;
+
+	public async run() {
+		const { args: params, flags: options } = this.parse<FlagsDef, ArgsDef>(
+			DeviceDeactivateCmd,
+		);
+
+		const balena = getBalenaSdk();
+		const patterns = await import('../../utils/patterns');
+
+		const uuid = params.uuid;
+		const deactivationWarning = `
+
+Warning! Deactivating a device will charge a fee equivalent to the
+normal monthly cost for the device (e.g. $1 for an essentials device);
+the device will not be charged again until it comes online.
+
+Are you sure you want to deactivate device ${uuid} ?`;
+
+		// Confirm
+		await patterns.confirm(!!options.yes, deactivationWarning);
+
+		// Deactivate
+		try {
+			const { id } = await balena.models.device.get(uuid, { $select: 'id' });
+			await balena.pine.patch<BalenaSdk.Device>({
+				resource: 'device',
+				body: {
+					is_active: false,
+				},
+				options: {
+					$filter: {
+						id,
+					},
+				},
+			});
+		} catch (err) {
+			if (err instanceof BalenaAmbiguousDevice) {
+				console.info(
+					`Error: Multiple devices found that their uuid starts with ${uuid}`,
```

Comment:
`that their` doesn't sound very good... Alternatives:
* `Error: Multiple devices found whose UUID starts with ${uuid}`,
* `Error: Multiple devices found with UUID starting with ${uuid}`,
* `Error: Multiple devices found of UUID starting with ${uuid}`,

Reply:
Also, you could have all CLI commands take advantage of this error handling and error message rewriting by doing it in `lib/errors.ts` instead of doing it in this file. Instead of catching the error here, let it bubble up to `errors.ts` (simply don't catch it here) and add a similar handling as for `BalenaExpiredToken` in the following lines:
https://github.com/balena-io/balena-cli/blob/v12.29.0/lib/errors.ts#L140-L142

It works because `BalenaAmbiguousDevice` defines an error code:
https://github.com/balena-io-modules/balena-errors/blob/v4.7.1/lib/errors.ts#L339

You would also add `BalenaAmbiguousDevice` to the following regex array:
https://github.com/balena-io/balena-cli/blob/v12.29.0/lib/errors.ts#L145-L146

The global error handling in `errors.ts` already sets `process.exitCode` to 1 by default:
https://github.com/balena-io/balena-cli/blob/v12.29.0/lib/errors.ts#L181-L184